lib/view_component/base.rb

```diff
@@ -23,6 +23,13 @@ class Base < ActionView::Base
     class_attribute :content_areas
     self.content_areas = [] # class_attribute:default doesn't work until Rails 5.2
 
+    # Hook for allowing components to do work as part of the compilation process.
+    #
+    # For example, one might compile component-specific assets at this point.
+    def self.after_compile
```

Comment:
I could also see this being nice as a DSL (similar to `after_filter`, `after_save`, etc), maybe something roughly like:

```ruby
class MyComponent < ViewComponent::Base
  after_compile :load_i18n
  
  private

  def load_i18n
    # find i18n files
  end
end
```

To avoid having to remember to call `super` each time you add more behavior, and the potential ability to opt-out of specific hooks like how `skip_before_action` works.